lighthouse-core/fraggle-rock/config/filters.js

```diff
@@ -45,7 +47,7 @@ function getAuditIdsInCategories(allCategories, onlyCategories) {
 
   onlyCategories = onlyCategories || Object.keys(allCategories);
   const categories = onlyCategories.map(categoryId => allCategories[categoryId]);
```

Comment:
Dangers of index signatures :) Because `allCategories` is `Record<string, Category>`, tsc doesn't warn that `categories` is actually `Array<categories|undefined>` here, since `categoryId` might not be a key of `allCategories`.

This is an intentional design decision of tsc (index signatures are too useful in js but not having exact types in ts makes it very painful to deal with them the "correct" way), though [`noUncheckedIndexedAccess`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-1.html#checked-indexed-accesses---nouncheckedindexedaccess) is always available to force having to always check for `undefined`. There's a reason it's not even part of `strict` though; it's pretty awful, especially with arrays.

Reply:
Yeah I've definitely been bitten by this before.